src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import styles from './style.scss';
+
+const defaultIndex = -1;
+const baseClass = 'btn-group';
+
+class ButtonGroup extends React.Component {
+  state = {
+    currentIndex: defaultIndex
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    return {
+      currentIndex:
+        typeof props.currentIndex === 'number'
+          ? props.currentIndex
+          : state.currentIndex
+    };
+  }
+
+  handleClick = index => () => {
+    this.props.onChange(index);
+    this.setState({ currentIndex: index });
+  };
+
+  render() {
+    const { currentIndex } = this.state;
+    const { size, fullWidth, children, ...restProps } = this.props;
+
+    const mappedChildren = React.Children.map(children, (child, index) =>
+      React.cloneElement(child, {
+        size,
+        fullWidth,
+        primary: false,
+        destructive: false,
+        onClick: this.handleClick(index),
+        className: index === currentIndex ? styles.active : null
```

Comment:
@walaszczykm look what happens when the middle button is active.
![image](https://user-images.githubusercontent.com/24878280/61031822-8c099f80-a3c0-11e9-8680-d76e29416d0e.png)

Reply:
I found a solution to this problem and implemented it 👍